src/mscorlib/corefx/System/Globalization/CalendarData.Unix.cs

```diff
@@ -1,36 +1,65 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
-using System;
-using System.Runtime.InteropServices;
-using System.Runtime.CompilerServices;
-using System.Diagnostics.Contracts;
+
 using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
 
 namespace System.Globalization
 {
+    // needs to be kept in sync with CalendarDataType in System.Globalization.Native
+    internal enum CalenderDataType
+    {
+        Uninitialized = 0,
+        NativeName = 1,
+        MonthDay = 2,
+        ShortDates = 3,
+        LongDates = 4,
+        YearMonths = 5,
+        DayNames = 6,
+        AbbrevDayNames = 7,
+        MonthNames = 8,
+        AbbrevMonthNames = 9,
+        SuperShortDayNames = 10,
+        MonthGenitiveNames = 11,
+        AbbrevMonthGenitiveNames = 12,
+        EraNames = 13,
+        AbbrevEraNames = 14,
+    }
+
     internal partial class CalendarData
     {
         private bool LoadCalendarDataFromSystem(String localeName, CalendarId calendarId)
         {
-            // TODO: Implement this.
-            return false;
+            bool result = true;
+            result &= GetCalendarInfo(localeName, calendarId, CalenderDataType.NativeName, out this.sNativeName);
```

Comment:
The issue is that we need to return a string[], not just a single string, and we don't necessarily know how many elements in the string array there will be (like for long and short date patterns).
This pattern has been used elsewhere - for example in the CalendarData.Win32 side - so @ellismg thought it would be best if we mirrored it.